agent/src/agent/destination/anodot_api_client.py

```diff
@@ -0,0 +1,79 @@
+import requests
+import urllib.parse
+import click
+
+from agent.destination import HttpDestination, AuthenticationToken
+from agent.modules import proxy
+from agent.modules.logger import get_logger
+from agent import destination
+
+logger = get_logger(__name__)
+
+
+def endpoint(func):
+    """
+    Logs errors and returns json response
+    """
+    def wrapper(*args, **kwargs):
+        args[0].refresh_session_authorization()
+        res = func(*args, **kwargs)
+        try:
+            res.raise_for_status()
+            return res.json()
+        except requests.exceptions.HTTPError:
+            if res.text:
+                logger.error(res.text)
+            raise
+    return wrapper
+
+
+class AnodotApiClientException(click.ClickException):
+    pass
+
+
+class AnodotApiClient:
+    def __init__(self, destination_: HttpDestination, use_authentication: bool = True):
+        self.url = destination_.url
+        self.access_key = destination_.access_key
+        self.proxies = proxy.get_config(destination_.proxy)
+        self.session = requests.Session()
+        if use_authentication:
```

Comment:
I don't think you need this flag. Just in case of failed validation don't save anything to the db